### ISSUE TITLE
Prerelease comparisons not according to semver.org

### ISSUE THREAD

**Issue by tazle** [k-bx/python-semver#21]
It seems that comparisons of pre-release versions do not work according semver.org specification:

Example:
```
$ python -c "import semver; print(semver.min_ver('1.2.3-rc2', '1.2.3-rc10'))"
1.2.3-rc2
```

According to semver.org, the parts after dash should be split at '.', and alphanumeric parts compared lexicographically and numeric parts numerically. It seems that this is not the case in semver-the-module.

**Comment by k-bx** [comment]
Thanks for spotting! Would you care to make a PR on this one?

**Comment by k-bx** [comment]
Released under 2.4.0
